library/src/main/java/com/chuckerteam/chucker/internal/support/NotificationHelper.kt

```diff
@@ -118,12 +118,18 @@ internal class NotificationHelper(val context: Context) {
                 context,
                 INTENT_REQUEST_CODE,
                 deleteIntent,
-                PendingIntent.FLAG_ONE_SHOT
+                PendingIntent.FLAG_ONE_SHOT or immutableFlag()
             )
             return NotificationCompat.Action(R.drawable.chucker_ic_delete_white, clearTitle, intent)
         }
 
     fun dismissNotifications() {
         notificationManager.cancel(TRANSACTION_NOTIFICATION_ID)
     }
+
+    private fun immutableFlag() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Do we really need this check?
Flags are generally forwards compatible.
During compilation constant will be inlined and in runtime on pre-M those bits won't be read.
Warning will be generated about using constants from higher API levels than min SDK but in this case such usage is legitimate IMHO.

Reply:
I'm not sure what are you proposing. I don't want IDE to highlight this and Lint to fail. Do you suggest to add `@SuppressLint("InlinedApi")`? R8 can remove this condition anyway with `-assumevalues`. Then there is dex2oat and finally ART itself.

Reply:
> Do you suggest to add @SuppressLint("InlinedApi")

Exactly.

> R8 can remove this condition

True, but not everyone use R8.

Reply:
> True, but not everyone use R8.

I would say that if someone does not use R8 then they don't care about optimizations anyway. :)

Reply:
I'd be more for having the explicit if-then-else. It's more aligned with the rest of the codebase and makes the code more explicit.